### PR TITLE
Form

### PR DESCRIPTION
**Pull request by ahmedadel56**
- [x] Create forms to perform the following functions:
- [x] Creates a Post on behalf of the current_user.
- [x] Create a comment on behalf of the current_user.
- [x] Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      post_id: @post.id
+    )
+    new_comment.update_comments_counter
+    if new_comment.save
+      redirect_to "/users/#{@post.user_id}/posts/#{@post.id}", notice: 'Success!'
+    else
+      render :new, alert: 'Error occured!'
+    end
+  end
+
```
Comment (rloterh) [inline app/controllers/comments_controller.rb:20]: - [x] _New_ action for _comments_ is missing. Kindly ensure to define _new_ action for _comments_ controller, similar to:
```ruby
   def new
     @comment = Comment.new
   end
```

File: config/routes.rb [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  root 'users#index'
+    resources :users, only: [:index, :show] do
+      resources :posts, only: [:index, :new, :create, :show]
+    end
 
-  # Defines the root path route ("/")
-end
+    resources :posts do
+      resources :comments, only: [:create]
```
Comment (rloterh) [inline config/routes.rb]: - [x] Kindly ensure that there is a pair of routes in the `config/routes.rb` file for creating comments like below:

```ruby
   POST   /posts/:post_id/comments(.:format)     comments#create
   GET   /posts/:post_id/comments/new(.:format)     comments#new
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Adel, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦   🚦  

## Good Points 👍  
- Descriptive Pull Request
- Good ReadMe
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bimbolabuari — APPROVED** [review]
# Project Approved 🚀

Hi @ahmedadel56🥇,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**